internal/mesh/internal/types/proxy_state_template.go

```diff
@@ -92,6 +93,20 @@ func ValidateProxyStateTemplate(res *pbresource.Resource) error {
 				}))
 			}
 
+			if portErr := catalog.ValidateProtocol(pbcatalog.Protocol(cluster.Protocol)); portErr != nil {
+				merr = multierror.Append(merr, wrapClusterErr(resource.ErrInvalidField{
+					Name:    "protocol",
+					Wrapped: portErr,
+				}))
+			}
+
+			if pbcatalog.Protocol(cluster.Protocol) == pbcatalog.Protocol_PROTOCOL_UNSPECIFIED {
+				merr = multierror.Append(merr, wrapClusterErr(resource.ErrInvalidField{
+					Name:    "protocol",
+					Wrapped: fmt.Errorf("cluster %q protocol is required but unspecified", name),
```

Comment:
There is `resource.ErrMissing` that we usually use as the Wrapped error in these cases. It's not mandatory I was just pointing it out.

Reply:
Ah, nice - thanks! Happy to swap.